frontend/app/helpers/useLoginValidation.ts

```diff
@@ -0,0 +1,73 @@
+import { useState } from "react"
+
+interface UserCredentials {
+  username: string
+  password: string
+}
+
+interface ValidationError {
+  username: string
+  password: string
+}
+
+const useLoginValidation = (): [UserCredentials, ValidationError, (event: React.ChangeEvent<HTMLInputElement>) => void, (event: React.ChangeEvent<HTMLFormElement>) => void] => {
+  const [credentials, setCredentials] = useState<UserCredentials>({
+    username: "",
+    password: ""
+  })
```

Comment:
I think it was mentioned to extract the default value outside.

Reply:
Thank you for your suggestion regarding extracting the default value outside of the useState call.
I understand your point and agree that it could improve readability and maintainability in this case. 
and i have done the necessary changes